### PR TITLE
fix(startup): run embedded Nvim with real path

### LINKED ISSUE #24277: Crash when changing argv0 with exec -a
### Problem

Running nvim with a different name like `exec -a foo nvim` fails with an error message:
~~~
E903: Process failed to start: no such file or directory: "foo"Segmentation fault (core dumped)
~~~
Running `exec -a nvim nvim` fails without an error message.

The error happens with the current master branch, 0.9.1 is not affected.

This was originally reported in https://github.com/lucc/nvimpager/issues/87

### Steps to reproduce

exec -a foo nvim
exec -a nvim nvim

### Expected behavior

Neovim should start normally. The command name in `ps -o cmd $PID` should be "foo".

### Neovim version (nvim -v)

v0.10.0-dev-811140e

### Vim (not Nvim) behaves the same?

n/a

### Operating system/version

Arch Linux & NixOS

### Terminal name/version

alacritty

### $TERM environment variable

alacritty

### Installation

AUR (package neovim-git), Nix Flake from this repo (contrib folder)

### PR DESCRIPTION
**Pull request by zeertzjq**
Fix #24277

### CODE REVIEW

File: src/nvim/channel.c [hunk 1]
```diff
@@ -322,6 +322,7 @@ static void close_cb(Stream *stream, void *data)
 ///                         stdin disconnected.
 /// @param[in]  cwd  Initial working directory for the job.  Nvim's working
 ///                  directory if `cwd` is NULL
+/// @param[in]  path  The path of the executable. If NULL, use `argv[0]`.
```
Comment (justinmk) [inline src/nvim/channel.c:325]:
```suggestion
/// @param[in]  arg0  The path of the executable. If NULL, use `argv[0]`.
```

It's probably helpful if this lives next to `argv`.

Comment (zeertzjq) [reply src/nvim/channel.c:325]: `arg0` is the name that makes the least sense. I think `exepath` or `cmd` are better.

**Comment by zeertzjq** [comment]
I tried to write a test for this using Python:
```lua
  it('argv[0] can be overridden when using TUI', function()
    skip(missing_provider('python3'), 'Missing python3 provider')
    finally(function()
      os.remove('Xoverride_argv0')
    end)
    exec([[
      python3 << trim EOF
          import os

          pid = os.fork()
          if pid == 0:
              os.execl(
                  vim.vvars["progpath"],
                  "foobar",
                  "--clean",
                  "-c",
                  'call writefile(v:argv + [v:progname], "Xoverride_argv0") | quit',
              )
          else:
              os.waitpid(pid, 0)
      EOF
    ]])
    eq(dedent([[
      foobar
      --clean
      -c
      call writefile(v:argv + [v:progname], "Xoverride_argv0") | quit
      foobar
      ]]), read_file('Xoverride_argv0'))
  end)
```
But it seems that the test always succeeds,

**Comment by zeertzjq** [comment]
Oh, TUI is only started when there is a PTY. `uv.spawn()` also does not work as it prepend `path` to `args`. I guess the only way to test this is to add this to `jobstart()` then, which should probably be done in another PR (if that should be done at all).

**Comment by justinmk** [comment]
> TUI is only started when there is a PTY. I guess the only way to test this is to add this to `termopen()` then

In `test/functional/terminal/tui_spec.lua` we start nvim in `:terminal` to test the TUI. Could check `exepath` by using `assert_log` https://github.com/neovim/neovim/blob/804108af6c79ae20d372a6c295ff292887d40c20/test/functional/options/defaults_spec.lua#L364

Side note: we should deprecate `termopen()` now that we have `nvim_open_term()`

**Comment by zeertzjq** [comment]
Oh wait, I think this can be tested using a Python script in `:terminal`.

**Comment by zeertzjq** [comment]
> Side note: we should deprecate `termopen()` now that we have `nvim_open_term()`

Doesn't that require #23747?

**Pull request merged**